framework/src/iotbus/iotapi_sig_handler.c

```diff
@@ -0,0 +1,152 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <debug.h>
+#include <errno.h>
+#include <signal.h>
+#include <iotbus/iotbus_gpio.h>
+#include <iotbus/iotbus_uart.h>
+
+#include "iotapi_sig_handler.h"
+
+#define IOTAPI_SIG_THREAD_SIZE	2048
+#define IOTAPI_THREAD_PRIORITY	120
+
+static pthread_addr_t iotapi_sig_work(pthread_addr_t arg);
+static pthread_t g_tid = -1;
+
+static pthread_t _create_sig_thread(void *arg)
+{
+	pthread_t tid;
+	pthread_attr_t attr;
+	struct sched_param sparam;
+	int ret = 1;
+
+	ret = pthread_attr_init(&attr);
+	if (ret != 0) {
+		ibdbg("failed to set pthread init(%d)\n", ret);
+		return -ret;
+	}
+
+	ret = pthread_attr_setschedpolicy(&attr, SCHED_RR);
+	if (ret != 0) {
+		ibdbg("failed to set policy(%d)\n", ret);
+		return -ret;
+	}
+
+	ret = pthread_attr_setstacksize(&attr, IOTAPI_SIG_THREAD_SIZE);
+	if (ret != 0) {
+		ibdbg("failed to set stack size(%d)\n", ret);
+		return -ret;
+	}
+
+	sparam.sched_priority = IOTAPI_THREAD_PRIORITY;
+	ret = pthread_attr_setschedparam(&attr, &sparam);
+	if (ret != 0) {
+		ibdbg("failed to set sched param(%d)\n", ret);
+		return -ret;
+	}
+
+	ret = pthread_create(&tid, &attr, iotapi_sig_work, arg);
+	if (ret != 0) {
+		ibdbg("failed to create pthread(%d)\n", ret);
+		return -ret;
+	}
+
+	pthread_setname_np(tid, "iotapi_handler");
+	pthread_detach(tid);
+
+	return tid;
+}
+
+static pthread_addr_t iotapi_sig_work(pthread_addr_t arg)
+{
+	void *ptr;
+	int started = 1;
+
+	sigset_t set;
+	siginfo_t info;
+
+	sigemptyset(&set);
+	sigaddset(&set, SIG_IOTBUS_GPIO_FALLING);
```

Comment:
I think good sigaddset(&set, SIG_IOTBUS_GPIO_FALLING | SIG_IOTBUS_GPIO_RISING .... | SIG_IOTBUS_UART_RECEIVED)

Reply:
In my opinion, it might need to separate the signal considering the priority of types of the interrupt. By merging all types of the signal cannot guarantee the parallel processing and the priority scheduling.